include/fmt/ranges.h

```diff
@@ -528,11 +528,27 @@ struct formatter<
 
  public:
   FMT_CONSTEXPR formatter() {
-    if (detail::const_check(range_format_kind<R, Char>::value !=
-                            range_format::set))
-      return;
-    range_formatter_.set_brackets(detail::string_literal<Char, '{'>{},
-                                  detail::string_literal<Char, '}'>{});
+    switch (range_format_kind<R, Char>::value) {
+    case range_format::debug_string: {
+      range_formatter_.set_brackets(detail::string_literal<Char, '"'>{},
+                                    detail::string_literal<Char, '"'>{});
+      range_formatter_.set_separator({});
+      break;
+    }
+    case range_format::set: {
+      range_formatter_.set_brackets(detail::string_literal<Char, '{'>{},
+                                    detail::string_literal<Char, '}'>{});
+      break;
+    }
+    case range_format::string: {
+      range_formatter_.set_brackets({}, {});
+      range_formatter_.set_separator({});
```

Comment:
Doesn't seem to work as expected. I have to look into it.

Reply:
I end up with something like:  '\'''A''\'''\'''s''\'''\'''s''\'''\'''e''\ and don't seem to go through the `range_formatter::format`.